src/components/TabBar/index.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+
+interface TabBarProps {
+  tabs: {
+    title: string
+    render: () => void
+  }[]
+}
+
+export default function TabBar({ tabs }: TabBarProps) {
+  const [activeTab, setActiveTab] = useState(0)
+
+  return (
+    <>
+      <TabContainer>
+        {tabs.map((tab, index) => (
+          <TabButton key={index} onClick={() => setActiveTab(index)} active={activeTab === index}>
+            <Title active={activeTab === index}>{tab.title}</Title>
+          </TabButton>
+        ))}
+      </TabContainer>
+      {tabs[activeTab].render()}
+    </>
+  )
+}
+
+const TabContainer = styled.section`
+  display: flex;
+  flex-direction: row;
+  width: 100%;
+  height: 50px;
+`
+
+const TabButton = styled.button<{ active: boolean }>`
+  width: auto;
+  cursor: pointer;
+  padding: 0;
+  border: none;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  position: relative;
+  transition: 0.6s;
```

Comment:
What is the transition on here? If it is on all, I'd advise reviewing this to work on specific attributes (even though I don't see any animatable attribute here?).